### PR TITLE
Acura RDX 2018 AcuraWatch Plus Port

### PR DESCRIPTION
**Pull request by vanillagorillaa**
Acura RDX 2018 AcuraWatch Plus Port. Just needs to be tested

### CODE REVIEW

File: selfdrive/car/honda/hondacan.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def create_accord_steering_control(apply_steer, idx):
 def create_steering_control(apply_steer, crv, idx):
   """Creates a CAN message for the Honda DBC STEERING_CONTROL."""
   commands = []
-  if crv:
+  if crv or rdx:
```
Comment (rbiasini) [inline selfdrive/car/honda/hondacan.py]: rdx isn't an input here

File: selfdrive/car/honda/carcontroller.py [hunk 2]
```diff
@@ -175,6 +175,6 @@ def update(self, sendcan, enabled, CS, frame, actuators, \
 
     if (frame % radar_send_step) == 0:
       idx = (frame/radar_send_step) % 4
-      can_sends.extend(hondacan.create_radar_commands(CS.v_ego, CS.civic, CS.accord, CS.crv, idx))
+      can_sends.extend(hondacan.create_radar_commands(CS.v_ego, CS.civic, CS.accord, CS.crv, CS.rdx, idx))
```
[the remark this answers is not in the document]
Comment (vanillagorillaa) [reply selfdrive/car/honda/carcontroller.py]: Should be set now

File: selfdrive/car/honda/carstate.py [hunk 3]
```diff
@@ -433,14 +487,28 @@ def update(self, cp):
...
+      self.cruise_buttons = cp.vl[0x1A6]['CRUISE_BUTTONS']
+      self.main_on = cp.vl[0x1A6]['MAIN_ON']
+      self.blinker_on = cp.vl[0x294]['LEFT_BLINKER'] or cp.vl[0x294]['RIGHT_BLINKER']
+      self.left_blinker_on = cp.vl[0x294]['LEFT_BLINKER']
+      self.right_blinker_on = cp.vl[0x294]['RIGHT_BLINKER']
+      self.cruise_speed_offset = -0.3
+      self.park_brake = 0  # TODO
+      self.brake_hold = 0  # RDX doesn't have EPB  
 
     self.gear_shifter = parse_gear_shifter(can_gear_shifter, self.acura)
```
Comment (rbiasini) [inline selfdrive/car/honda/carstate.py]: so, rdx gear shifter is parsed as the civic?

Comment (vanillagorillaa) [reply selfdrive/car/honda/carstate.py]: RDX matches civic, 0x8-D, 0x1-P, 0x2-R, 0x4-N

**Comment by vanillagorillaa** [comment]
@rbiasini This should be good for alpha support if we can still get it in for 0.4.0. Still going to work on Steer_Max and will submit a PR at a later point once we find final value

**Comment by vanillagorillaa** [comment]
Cannot forget to give huge credit to @energee for his help and support on the port!

**Comment by rbiasini** [comment]
This is great work guys! Will PM @vanillagorillaa  and @energee about the TODO list for the merge. It's pretty close.

**Comment by energee** [comment]
@rbiasini The person testing is relaying that awareness is not being reset by a steer-override or button events like cruise reset or resume. I can't see how this is possible especially since it is clearly reading the buttons because openpilot is being engaged. I am not seeing any obvious reason for this and theres not too much I can do without the ability to debug, any ideas?

**Comment by rbiasini** [comment]
@energee I believe that steer override might not reset awareness status because the steer torque sensor is probably wrong in the dbc. It's strange about the buttons though. Is he sure?

**Comment by energee** [comment]
I checked steer torque sensor, I think it is ok. 
screen shot 2017-12-21 at 4 31 13 pm
Was hoping there was something obvious i was missing. It is hard doing this without physical access!

**Comment by vanillagorillaa** [comment]
Another thing that is happening is every time he starts up the car is that OP is not giving him the "Warning, keep hands on the wheel" message. He says this occurs every time on start up. He is not 100% sure that changing speed is resetting timer so hes going to look tonight. Can confirm that OP disengages when Gas/Break/Cancel is applied

**Comment by vanillagorillaa** [comment]
@rbiasini Solved steer_torque_sensor issue. RDX uses a lower value than other Honda's. This value gives us same behavior as other Honda's and resets awareness timer appropriately. Doing Max Brake test tomorrow and this should be done and good

**Comment by dbergan** [comment]
Awesome! Anyone have info on how similar RDX is to the MDX? Easy port?

**Comment by vanillagorillaa** [comment]
@dbergan Which year MDX? I beileve that the MDX uses LIN for steering which nobody has worked on yet.

**Comment by dbergan** [comment]
Looking at the 2018 Acura MDX or RDX as a new car for me. For my wife we're wanting a "big" SUV like a GMC Yukon XL, Lincoln Navigator, Cadillac Escalade, etc.  It doesn't sound like there are any big SUVs that are close to being OpenPilot compatible (the Sequoia lacks steering assist... and Honda/Acura doesn't make one).

**Comment by Joker2k999** [comment]
How about an 2014 Acura MDX w/ the Advance & tech package?  same/justAsEz as RDX to port?

**Comment by vanillagorillaa** [comment]
@Joker2k999 See above comments about MDX.

File: selfdrive/car/honda/carstate.py [hunk 4]
```diff
@@ -432,6 +486,20 @@ def update(self, cp):
...
+      self.angle_steers = cp.vl[0x156]['STEER_ANGLE']
+      self.angle_steers_rate = cp.vl[0x156]['STEER_ANGLE_RATE']
+      self.gear = cp.vl[0x188]['GEAR']
+      self.cruise_setting = cp.vl[0x1A6]['CRUISE_SETTING']
+      self.cruise_buttons = cp.vl[0x1A6]['CRUISE_BUTTONS']
+      self.main_on = cp.vl[0x1A6]['MAIN_ON']
+      self.blinker_on = cp.vl[0x294]['LEFT_BLINKER'] or cp.vl[0x294]['RIGHT_BLINKER']
+      self.left_blinker_on = cp.vl[0x294]['LEFT_BLINKER']
+      self.right_blinker_on = cp.vl[0x294]['RIGHT_BLINKER']
+      self.cruise_speed_offset = -0.3
```
Comment (rbiasini) [inline selfdrive/car/honda/carstate.py]: here we should use CRUISE_SPEED_OFFSET instead of hardcoding -0.3

**Comment by vanillagorillaa** [comment]
Should also be added that OpenPilot cannot be engaged under 25mph

File: selfdrive/car/honda/carcontroller.py [hunk 5]
```diff
@@ -121,7 +121,7 @@ def update(self, sendcan, enabled, CS, frame, actuators, \
     if CS.CP.carFingerprint in (CAR.CIVIC, CAR.ODYSSEY):
       is_fw_modified = os.getenv("DONGLE_ID") in ['b0f5a01cf604185c']
       STEER_MAX = 0x1FFF if is_fw_modified else 0x1000
-    elif CS.crv:
+    elif CS.crv or CS.acura_rdx:
```
Comment (rbiasini) [inline selfdrive/car/honda/carcontroller.py]: this is a bug. Actually the same bug that broke that was not civic or odyssey. replace by:
```elif CS.CP.carFingerprint in (CAR.CRV, CAR.ACURA_RDX):```

Comment (vanillagorillaa) [reply selfdrive/car/honda/carcontroller.py]: Fixed

File: selfdrive/car/honda/carstate.py [hunk 6]
```diff
@@ -235,19 +235,24 @@ def update(self, cp):
...
+   
       self.main_on = cp.vl["SCM_BUTTONS"]['MAIN_ON']
 
     self.cruise_speed_offset = calc_cruise_offset(cp.vl["CRUISE_PARAMS"]['CRUISE_SPEED_OFFSET'], self.v_ego)
     self.gear_shifter = parse_gear_shifter(can_gear_shifter, self.CP.carFingerprint)
 
     self.pedal_gas = cp.vl["POWERTRAIN_DATA"]['PEDAL_GAS']
     # crv doesn't include cruise control
-    if self.CP.carFingerprint != CAR.CRV:
+    if self.CP.carFingerprint != CAR.CRV or CAR.ACURA_RDX:
```
Comment (rbiasini) [inline selfdrive/car/honda/carstate.py]: @vanillagorillaa this should be 
```if CS.CP.carFingerprint not in (CAR.CRV, CAR.ACURA_RDX)```

actually, I would invert the check as follow:
```
if CS.CP.carFingerprint in (CAR.CRV, CAR.ACURA_RDX):
  self.car_gas = self.pedal_gas
else:
  self.car_gas = cp.vl["GAS_PEDAL_2"]['CAR_GAS']

File: selfdrive/car/honda/hondacan.py [hunk 7]
```diff
@@ -49,11 +49,10 @@ def create_gas_command(gas_amount, idx):
   msg = struct.pack("!H", gas_amount)
   return make_can_msg(0x200, msg, idx, 0)
 
-
 def create_steering_control(apply_steer, car_fingerprint, idx):
   """Creates a CAN message for the Honda DBC STEERING_CONTROL."""
   commands = []
-  if car_fingerprint == CAR.CRV:
+  if car_fingerprint == CAR.CRV or CAR.ACURA_RDX:
```
Comment (rbiasini) [inline selfdrive/car/honda/hondacan.py]: bug

**Comment by voland696** [comment]
Acura RDX JAE connector pinout? Same like JAE column in the guide?

**Pull request merged**